docs/design/pattern_matching.md

```diff
@@ -387,10 +368,13 @@ match ({.a = 1, .b = 2}) {
 }
 ```
 
-The scrutinee is required to be of struct type, and to have the same set of
-field names as the pattern. The pattern is matched left-to-right, meaning that
-matching is performed in the field order specified in the pattern, not in the
-field order of the scrutinee. This is consistent with the behavior of matching
+The scrutinee is expected to have a [struct form](values.md#expression-forms)
+with the same set of field names as the pattern, and form decomposition (but no
+other conversion) is applied if necessary to satisfy that. Then, each subpattern
+of the struct pattern in left-to-right order is matched with `s.f`, where `s` is
+the converted scrutinee expression and `f` is the field name associated with the
+subpattern. The struct pattern matches if all of these sub-matches succeed.
+Matching in left-to-right order is consistent with the behavior of matching
```

Comment:
Awkward phrasing. I think you want to say matching is done for fields in left-to-right order. You should be explicit about whether that is the left-to-right order of the fields in the scrutinee or the pattern.